pkg/client-sdk/covenantless_client.go

```diff
@@ -329,6 +334,155 @@ func (a *covenantlessArkClient) listenForArkTxs(ctx context.Context) {
 	}
 }
 
+func (a *covenantlessArkClient) refreshTxDb(ctx context.Context) error {
+	spendableVtxos, spentVtxos, err := a.ListVtxos(ctx)
+	if err != nil {
+		return err
+	}
+
+	boardingTxs, roundsToIgnore, err := a.getBoardingTxs(ctx)
+	if err != nil {
+		return err
+	}
+
+	offchainTxs, err := vtxosToTxsCovenantless(spendableVtxos, spentVtxos, roundsToIgnore)
+	if err != nil {
+		return err
+	}
+
+	oldTxs, err := a.store.TransactionStore().GetAllTransactions(ctx)
+	if err != nil {
+		return err
+	}
+
+	if len(oldTxs) == 0 {
+		if _, err := a.store.TransactionStore().AddTransactions(ctx, offchainTxs); err != nil {
+			return err
+		}
+
+		vtxosToAdd := make([]types.Vtxo, 0, len(spendableVtxos))
+		for _, vtxo := range spendableVtxos {
+			vtxosToAdd = append(vtxosToAdd, types.Vtxo{
+				VtxoKey: types.VtxoKey{
+					Txid: vtxo.Txid,
+					VOut: vtxo.VOut,
+				},
+				PubKey:    vtxo.PubKey,
+				Amount:    vtxo.Amount,
+				RoundTxid: vtxo.RoundTxid,
+				ExpiresAt: vtxo.ExpiresAt,
+				CreatedAt: vtxo.CreatedAt,
+			})
+		}
+		if _, err := a.store.VtxoStore().AddVtxos(ctx, vtxosToAdd); err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	oldTxsMap := make(map[string]types.Transaction, len(oldTxs))
+	for _, tx := range oldTxs {
+		oldTxsMap[tx.TransactionKey.String()] = tx
+	}
+
+	// add new boarding, update existing that are not confirmed
+	boardingTxsToAdd := make([]types.Transaction, 0)
+	boardingTxsToConfirm := make([]types.Transaction, 0)
+	for _, tx := range boardingTxs {
+		oldTx, exists := oldTxsMap[tx.Hex]
+		if !exists {
+			boardingTxsToAdd = append(boardingTxsToAdd, tx)
+		} else {
+			if oldTx.CreatedAt.IsZero() && !tx.CreatedAt.IsZero() {
+				boardingTxsToConfirm = append(boardingTxsToConfirm, tx)
+			}
```

Comment:
we also need to keep track of the txs to settle: if an old tx is not settled and the new one is, we should add it to the list `boardingTxsToSettle`.

There's an edge case here: we may need to take care of the case where a tx is unconfirmed in db, while the one "fetched form the explorer" is settled. Such txs needs to have updated both the CreatedAt and Settled fields, the event produced for this is ok to be tx_settled.

Reply:
> There's an edge case here: we may need to take care of the case where a tx is unconfirmed in db, while the one "fetched form the explorer" is settled. Such txs needs to have updated both the CreatedAt and Settled fields, the event produced for this is ok to be tx_settled.

we dont have repo method to update both fileds, shuold we create it? Smtn like UpdateTx ?